src/base/Formatter.ml

```diff
@@ -0,0 +1,514 @@
+(*
+  This file is part of scilla.
+
+  Copyright (c) 2018 - present Zilliqa Research Pvt. Ltd.
+
+  scilla is free software: you can redistribute it and/or modify it under the
+  terms of the GNU General Public License as published by the Free Software
+  Foundation, either version 3 of the License, or (at your option) any later
+  version.
+
+  scilla is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+  A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along with
+  scilla.  If not, see <http://www.gnu.org/licenses/>.
+*)
+
+(** Scilla code formatter *)
+
+[@@@ocamlformat "disable"]
+
+open Base
+open PPrint
+
+module Format (Lit : Literal.ScillaLiteral) =
```

Comment:
Could we parametrize `ER` and `SR` here which are used in instantiation of `ScillaSyntax` below?

Something like:
```ocaml
module Format (ER : Rep) (SR : Rep) (Lit : Literal.ScillaLiteral) =
```

Otherwise it cannot be used with `ER` and `SR` we got after typechecking.

For example, it is not possible to format the code in the `Checker` module which uses `OutputERep` and `OutputSRep`:

https://github.com/Zilliqa/scilla/blob/2509ee7e96527d88875ca1898584c6de24426bbc/src/base/Checker.ml#L51-L52

The error will be something like:

```
Error: This expression has type MProduct.PSyntax.cmodule
       but an expression was expected of type
         Formatter.GlobalLiteralSyntax.Ast.cmodule
```

Which says that the `ScillaSyntax` instantiated with the given `ER` and `SR` is incompatible with the `Formatter.Ast`